### PR TITLE
Integration: Fixes for parity replay

### PR DESCRIPTION
**Pull request by jmcardon**
PR checklist:

* [ ] Test coverage for the proposed changes
* [ ] PR description contains example output from repl interaction or a snippet from unit test output
* [ ] (If Relevant) Documentation has been (manually) updated at https://docs.kadena.io/pact

Additionally, please justify why you should or should not do the following:

* [ ] Benchmark regressions
* [x] Confirm replay/back compat (Ignore until core release)
* [x] (For Kadena engineers) Run integration-tests against a Chainweb built with this version of Pact (Ignore until core release)

### CODE REVIEW

File: pact/Pact/Core/IR/Term.hs [hunk 1]
```diff
@@ -511,28 +520,30 @@ instance (Pretty name, Pretty builtin, Pretty ty) => Pretty (Term name ty builti
...
     TLTerm tm -> pretty tm
-    _ -> "todo: pretty defs/modules"
-
-prettyDef :: Pretty ty => Doc ann -> Arg ty info -> info -> [Arg ty info] -> Doc ann
-prettyDef deftoken (Arg defname defRTy _) dInfo defArgs =
-  let dfNameArg = Arg defname defRTy dInfo
-      argList = parens (hsep (pretty <$> defArgs))
-  in parens $ deftoken <+> pretty dfNameArg <+> argList
+    TLModule m -> pretty m
+    _ -> "todo: pretty interfaces"
```
Comment (emilypi) [inline pact/Pact/Core/IR/Term.hs:531]: Is there ever an instance where we'll need to prettyprint an interface?

**Review by emilypi — APPROVED** [review]

File: pact/Pact/Core/IR/Term.hs (hunk 1 above)
Comment (jmcardon) [reply pact/Pact/Core/IR/Term.hs:531]: Probably yes for debugging. Thanks for the note, I'll remember to add pretty instances.

For transparency: These `pretty` instances are being used during integration work to debug the pact-5 representation of the modules read in from legacy. They're incredibly useful to see how we are deserializing things, but only interfaces presented a challenge.

File: pact/Pact/Core/Gas/Utils.hs [hunk 2]
```diff
@@ -60,3 +62,21 @@ chargeFlatNativeGas :: i -> b -> EvalM e b i ()
...
+-- this function assumes gas logs are in order
+prettyGasLogs :: Show b => GasModel b -> [GasArgs b] -> T.Text
+prettyGasLogs model = \case
+  [] -> "TOTAL: 0"
+  li -> let
+    (str, total) = foldl' go ([], MilliGas 0) li
+    in "TOTAL: " <> T.pack (show total) <> "\n" <> T.unlines (reverse str)
+  where
+  prettyLine (ga, used, amt) =
+    -- let used = runTableModel (_gmNativeTable model) (_gmGasCostConfig model) ga
```
Comment (rsoeldner) [inline pact/Pact/Core/Gas/Utils.hs:75]: cleanup

File: pact/Pact/Core/IR/Eval/CEK/CoreBuiltin.hs [hunk 3]
```diff
@@ -123,8 +123,9 @@ roundingFn :: (IsBuiltin b) => (Rational -> Integer) -> NativeFunction e b i
 roundingFn op info b cont handler _env = \case
   [VLiteral (LDecimal d)] ->
     returnCEKValue cont handler (VLiteral (LInteger (truncate (roundTo' op 0 d))))
-  [VDecimal d, VInteger prec] ->
-    returnCEKValue cont handler (VLiteral (LDecimal (roundTo' op (fromIntegral prec) d)))
+  [VDecimal d, VInteger prec] -> do
+    let roundPrec = max 0 (fromIntegral prec)
```
Comment (rsoeldner) [inline pact/Pact/Core/IR/Eval/CEK/CoreBuiltin.hs:127]: 👍

File: pact/Pact/Core/SizeOf.hs [hunk 4]
```diff
@@ -217,7 +217,7 @@ instance SizeOf SBS.ShortByteString where
 
 instance SizeOf Text where
   estimateSize t =
-    -- We will
+    -- We will rock you
```
Comment (rsoeldner) [inline pact/Pact/Core/SizeOf.hs:220]: <3

**Review by rsoeldner — APPROVED** [review]

**Pull request merged**